queries.sql

```diff
@@ -8,4 +8,43 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR 'Pikachu';
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+/* part 2 */
+
+BEGIN:
+UPDATE animals SET species = 'unspecified';
+SELECT species FROM animals;
+ROLLBACK;
+SELECT species FROM animals;
+
+
+
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species = NULL;
+COMMIT;
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT SAVE_POINT_1_ANIMALS;
+UPDATE animals SET weight_kg = (-1 * weight_kg);
+ROLLBACK TO SAVEPOINT SAVE_POINT_1_ANIMALS;
+UPDATE animals SET weight_kg = (-1 * weight_kg) WHERE weight_kg < 0;
+COMMIT;
+
+
+
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT * FROM animals WHERE neutered = true OR neutered = false GROUP BY neutered;
+
+SELECT neutered, MIN(weight_kg), MAX(weight_kg) FROM animals GROUP BY neutered;
+SELECT AVG(escape_attempts), neutered FROM animals WHERE date_of_birth > '1990-01-01' AND date_of_birth < '2000-12-01' GROUP BY neutered ;
```

Comment:
* I think you can select animals grouped by species for this question ( What is the average number of escape attempts per animal type of those born between 1990 and 2000?)  for example this query targets a particular animal type from the animal table where the date of birth is between 1997 to 2021 `SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals GROUP BY species;  WHERE  date_of_birth >= '1997/01/01' AND date_of_birth <= '2021/31/12'`
* This is because the species column is used to specify the animal type 
* For this question ( What is the average number of escape attempts per animal type of those born between 1990 and 2000?) you should add `SELECT species and then group by species` to your answer this way your query is targeting a specific animal type whose birthdate falls between `1990 to 2000`  
* Use this hint to answer all query related to animal type

Reply:
Thank you for the explanation.